openstates/dc/bills.py

```diff
@@ -1,212 +1,449 @@
+
+                bill.add_source(bill_source_url)
+                self.save_bill(bill)
+            
+            #get next page
+            start_record += per_page
+            params["request"]["iDisplayStart"] = start_record
+            param_json = json.dumps(params)
+            response = self.post(url,headers=headers,data=param_json)
+            response = self.decode_json(response.json()["d"])
+            data = response["aaData"]
+    
+    def get_member_ids(self):
+        member_dict = {} #three levels: from session to member_id to name
+        search_data_url = "http://lims.dccouncil.us/_layouts/15/uploader/AdminProxy.aspx/GetPublicSearchData"
+        response = self.post(search_data_url,headers={"Content-Type":"application/json"})
+        member_data = self.decode_json(response.json()['d'])["Members"]
+        for session_id, members in member_data.items():
+            member_dict[session_id] = {}
+            for member in members:
+                member_id = int(member["ID"]) #should be ints already, but once one wasn't
+                member_name = member["MemberName"]
+                member_dict[session_id][member_id] = member_name
+
+        return member_dict
+
+
+    def process_vote(self, vote, bill, member_ids):
+        try:
+            motion = vote["ReadingDescription"]
+        except KeyError:
+            self.logger.warning("Can't even figure out what we're voting on. Skipping.")
+            return
+
+        if not "VoteResult" in vote:
+            if "postponed" in motion.lower():
+                result = "Postponed"
+                status = True #because we're talking abtout the motion, not the amendment
+            elif "tabled" in motion.lower():
+                result = "Tabled"
+                status = True
+            else:
+                self.logger.warning("Could not find result of vote, skipping.")
+                return
+        else:
+
+            result = vote["VoteResult"]    
+
+            statuses = {"approved":True,
+                    "disapproved":False,
+                    "failed":False,
+                    "declined":False,
+                    "passed":True}
+
             try:
-                self.scrape_vote(bill, vote_type_id, vote_type)
-            except scrapelib.HTTPError as e:
-                self.warning(str(e))
+                status = statuses[result.strip().lower()]
+            except KeyError:
+                self.logger.warning("Unexpected vote result '{result},' skipping vote.".format(result=result))
+                return
+
+        date = self.date_format(vote["DateOfVote"])
+
+        leg_votes = vote["MappedJASON"] #their typo, not mine
+        v = Vote('upper',date,motion,status,0,0,0,
+                yes_votes=[],no_votes=[],other_votes=[])
+        for leg_vote in leg_votes:
+            mem_name = member_ids[int(leg_vote["MemberId"])]
+            if leg_vote["Vote"] == "1":
+                v['yes_count'] += 1
```

Comment:
OK, I lied about this, it can't be done. It *looks* like this should be available, but it's not in the underlying "JSON". They seem to be doing this same math on the fly for full council votes and hard-coding counts for committee votes. Ick!